scipy/stats/stats.py

```diff
@@ -3294,26 +3294,31 @@ def spearmanr(a, b=None, axis=0, nan_policy='propagate'):
     """
     a, axisout = _chk_asarray(a, axis)
 
-    contains_nan, nan_policy = _contains_nan(a, nan_policy)
+    a_contains_nan, nan_policy = _contains_nan(a, nan_policy)
 
-    if contains_nan and nan_policy == 'omit':
+    if a_contains_nan and nan_policy == 'omit':
         a = ma.masked_invalid(a)
-        b = ma.masked_invalid(b)
```

Comment:
I think that additional if statement isn't the best solution. 
If we have first parameter with `np.nan` element and second parameter without `np.nan` element, second parameter will be masked such as first parameter(it isn't necessary).